synapse/datamodel.py

```diff
@@ -873,10 +873,42 @@ def addForm(self, formname, forminfo, propdefs):
         for ifname in form.type.info.get('interfaces', ()):
             self._addFormIface(form, ifname)
 
+        self._checkFormDisplay(form)
+
         self.formprefixcache.clear()
 
         return form
 
+    def _checkFormDisplay(self, form):
+
+        formtype = self.types.get(form.full)
+
+        display = formtype.info.get('display')
+        if display is None:
+            return
+
+        for column in display.get('columns', ()):
+            coltype = column.get('type')
+            colopts = column.get('opts')
+
+            if coltype == 'prop':
```

Comment:
are we purposely omitting a way to specify a column for the primary prop?

Reply:
Not purposely, the first few definitions just don't require it. If the primary property would be useful, it's usually a sane default for a single-column view  :+1:

Reply:
Do you want to agree on a convention for it now so the optic changes can account for it?